src/tests/upgrades/red.ts

```diff
@@ -1,11 +1,56 @@
 import { expect } from "chai";
 
+import { GameEngine } from "../../framework/engine";
+import { createInstance } from "../../framework/mutators";
+import { InstanceState, PlayerState } from "../../framework/types";
+import { requireActivePlayer } from "../../game/helpers";
+
+import {
+  debugAction,
+  debugGotoNextTurn,
+  makeDefaultGame,
+} from "../testhelper";
+
 describe("upgrades", () => {
   describe("red", () => {
+    let $: GameEngine;
+    let P: PlayerState;
+
+    beforeEach(() => {
+      $ = makeDefaultGame();
+      P = requireActivePlayer($);
+    });
+
     describe("Bloodburn", () => {
+      let I: InstanceState;
+
+      beforeEach(() => {
+        I = createInstance($, P, $.data.lookupCard("Bloodburn"));
+      });
+
       it("can't be used on its first turn because it doesn't have " +
         "haste", () => {
-        expect(true).to.equal(false);
+        expect(I.arrivalFatigue).to.equal(true);
+
+        I.customRunes.BLOOD = 2;
+
+        expect(
+          () => debugAction($, {
+            type: "ACTIVATE_ABILITY",
+            instance: I.id,
+            ability: "Bloodburn #1",
```

Comment:
I've been wondering this since the first commit... Why do we need an arbitrary index to differentiate activatable abilities? I kinda would prefer I.id, I think.

Reply:
`I.id` isn't enough on its own, if that instance has multiple active abilities available. So I'm not sure I understand what you mean?